core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2006,15 +1997,11 @@ public static long toLong(Timestamp v) {
     return toLong(v, LOCAL_TZ);
   }
 
-  // mainly intended for java.sql.Timestamp but works for other dates also
-  @SuppressWarnings("JavaUtilDate")
-  public static long toLong(java.util.Date v, TimeZone timeZone) {
-    final long time = v.getTime();
-    return time + timeZone.getOffset(time);
+  public static long toLong(Timestamp v, TimeZone timeZone) {
```

Comment:
I updated the javadoc on the method in `SqlFunctions` that I touched. Most of it is a copy & paste from the methods I added to `DateTimeUtils`, but I thought it would be useful to re-explain it here too.

Reply:
Yes, it's a bit unfortunate to have to copy&paste but it's good to have the proper context (or pointers to where to get it).